active-rfcs/0000-build-performance.md

```diff
@@ -0,0 +1,187 @@
+- Start Date: 2021-12-13
+- Reference Issues:
+  - Previous: https://github.com/withastro/rfcs/pull/44
+- Implementation PR: https://github.com/withastro/astro/pull/2168
+
+# Summary
+
+In order to improve build performance and scale to sites that build thousands of pages, deprecate dynamic features that forces our build to scan rendered HTML and provide static replacements.
+
+These features will be required for SSR support as well, so this RFC seeks to unify how Astro sites are built in a way that is workable both for static-site generation (SSG) and server-side rendering (SSR).
+
+# Example
+
+The following features are bundled as part of this RFC:
+
+- Enforced static use of `client:` directives.
+- Deprecation of `Astro.resolve`.
+- Introduction of `local:` directive to act as a static replacement for Astro.resolve.
+
+## Static client hydration directives
+
+The directives used for client hydration are already static by implementation; if you attempt to add `client:load`, etc, during rendering Astro will not build that component's client-side JavaScript.
+
+This change will make this behavior defined and enforced. You can achieve dynamic usage by making the attribute static, but dynamically render a hydrated component like so:
+
+```astro
+---
+import Clock from '../components/Clock.jsx';
+const { shouldHydrate } = Astro.props;
+---
+{ shouldHydrate ? <Clock client:idle /> : <Clock /> }
```

Comment:
+1 This feels good for now, and would be an acceptable workaround for a final Astro v1.0 release